### PR TITLE
H-1068: Expose checking 'can user edit account group members' via Graph API

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

we need it to determine whether a user can add/remove org members

## 🔗 Related links

- H-1068

## 🚫 Blocked by

- #3394 

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

File: apps/hash-graph/lib/graph/src/api/rest/account.rs [hunk 1]
```diff
@@ -51,18 +73,34 @@ impl RoutedResource for AccountResource {
...
+                            )
+                            .nest(
+                                "/permissions",
+                                Router::new()
+                                    .route("/add_owner", get(can_add_group_owner::<A>))
+                                    .route("/remove_owner", get(can_remove_group_owner::<A>))
+                                    .route("/add_admin", get(can_add_group_admin::<A>))
+                                    .route("/remove_admin", get(can_remove_group_admin::<A>))
+                                    .route("/add_member", get(can_add_group_member::<A>))
+                                    .route("/remove_member", get(can_remove_group_member::<A>)),
```
Comment (TimDiekmann) [inline apps/hash-graph/lib/graph/src/api/rest/account.rs:102]: The idea is to pass a permission instead which can be checked but the types are not encoded correctly. I'm going to change this after the issue described in https://github.com/hashintel/hash/pull/3376 is solved.

File: apps/hash-graph/lib/graph/src/api/rest/account.rs [hunk 2]
```diff
@@ -162,6 +212,50 @@ where
...
+            .map_err(|error| {
+                tracing::error!(?error, "Could not acquire access to the authorization API");
+                StatusCode::INTERNAL_SERVER_ERROR
+            })?
+            .can_add_group_owner(actor_id, account_group_id, Consistency::FullyConsistent)
+            .await
+            .map_err(|error| {
+                tracing::error!(
+                    ?error,
+                    "Could not check if group member could be added by the specified actor"
```
Comment (CiaranMn) [inline apps/hash-graph/lib/graph/src/api/rest/account.rs:251]:
```suggestion
                    "Could not check if group owner could be added by the specified actor"
```

File: apps/hash-graph/lib/graph/src/api/rest/account.rs [hunk 3]
```diff
@@ -219,6 +313,50 @@ where
...
+            .map_err(|error| {
+                tracing::error!(?error, "Could not acquire access to the authorization API");
+                StatusCode::INTERNAL_SERVER_ERROR
+            })?
+            .can_remove_group_owner(actor_id, account_group_id, Consistency::FullyConsistent)
+            .await
+            .map_err(|error| {
+                tracing::error!(
+                    ?error,
+                    "Could not check if group member could be removed by the specified actor"
```
Comment (CiaranMn) [inline apps/hash-graph/lib/graph/src/api/rest/account.rs:352]:
```suggestion
                    "Could not check if group owner could be removed by the specified actor"
```

File: apps/hash-graph/lib/graph/src/api/rest/account.rs [hunk 4]
```diff
@@ -276,6 +414,50 @@ where
...
+            .map_err(|error| {
+                tracing::error!(?error, "Could not acquire access to the authorization API");
+                StatusCode::INTERNAL_SERVER_ERROR
+            })?
+            .can_add_group_admin(actor_id, account_group_id, Consistency::FullyConsistent)
+            .await
+            .map_err(|error| {
+                tracing::error!(
+                    ?error,
+                    "Could not check if group member could be added by the specified actor"
```
Comment (CiaranMn) [inline apps/hash-graph/lib/graph/src/api/rest/account.rs:453]:
```suggestion
                    "Could not check if group admin could be added by the specified actor"
```

File: apps/hash-graph/lib/graph/src/api/rest/account.rs [hunk 5]
```diff
@@ -333,6 +515,50 @@ where
...
+            .map_err(|error| {
+                tracing::error!(?error, "Could not acquire access to the authorization API");
+                StatusCode::INTERNAL_SERVER_ERROR
+            })?
+            .can_remove_group_admin(actor_id, account_group_id, Consistency::FullyConsistent)
+            .await
+            .map_err(|error| {
+                tracing::error!(
+                    ?error,
+                    "Could not check if group member could be removed by the specified actor"
```
Comment (CiaranMn) [inline apps/hash-graph/lib/graph/src/api/rest/account.rs:554]:
```suggestion
                    "Could not check if group admin could be removed by the specified actor"
```

**Review by CiaranMn — COMMENTED** [review]
Looks good, just a few suggestions on error messages to avoid any confusion

**Comment by TimDiekmann** [comment]
Thanks, @CiaranMn, I committed the changes 🙂

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**